src/runtime/Types/ClassObject.cs

```diff
@@ -48,6 +48,43 @@ internal NewReference GetDocString()
         }
 
 
+        /// <summary>
+        /// given an enum, write a __repr__ string formatted in the same
+        /// way as a python repr string. Something like:
+        ///   '&lt;Color.GREEN: 2&gt;';
+        /// with a binary value for [Flags] enums
+        /// </summary>
+        /// <param name="inst">Instace of the enum object</param>
+        /// <returns></returns>
+        private static string getEnumReprString(object inst)
+        {
+            var obType = inst.GetType();
+
+            var isFlags = obType.IsFlagsEnum();
+            var intValue = Convert.ToInt32(inst);
+            var intStr = isFlags ? "0x" + intValue.ToString("X4") : intValue.ToString();
```

Comment:
`X4` is insufficient for `Int32`, `X8` is required.
IMHO, it's best to set X according to the underlying integer type. Then it will immediately be clear from the `repr` what integer type backs the enum.